subprojects/language-java/src/integTest/groovy/org/gradle/java/compile/incremental/AbstractSourceIncrementalCompilationIntegrationTest.groovy

```diff
@@ -45,6 +45,60 @@ abstract class AbstractSourceIncrementalCompilationIntegrationTest extends Abstr
         outputs.recompiledClasses 'B', 'C'
     }
 
+    class LibWithAnonymousClasses {
+        void writeToProject() {
+            source """
+                interface SomeInterface {
+                    int foo();
+                }
+            """
+            source """
+                class Anonymous1 {
+                    SomeInterface getAnonymous() {
+                        return new SomeInterface() {
+                            public int foo() {
+                                return 0;
+                            }
+                        };
+                    }
+                }
+            """
+            source """
+                class Anonymous2 {
+                    SomeInterface getAnonymous() {
+                        return new SomeInterface() {
+                            public int foo() {
+                                return 1;
+                            }
+                        };
+                    }
+                }
+            """
+            source """
+                class Consumer {
+                    void consume() {
+                        System.out.println(new Anonymous1().getAnonymous().foo());
+                        System.out.println(new Anonymous2().getAnonymous().foo());
+                    }
+                }
+            """
+        }
+
+        void modifyAnonymousImplementation() {
+            source """
+                class Anonymous2 {
+                    SomeInterface getAnonymous() {
+                        return new SomeInterface() {
+                            public int foo() {
+                                return 42;
+                            }
+                        };
+                    }
+                }
+            """
```

Comment:
Do we need to check for anonymous classes using method-level nested classes?

```
class Anonymous2 {
    SomeInterface getAnonymous() {
        class Foobar implements SomeInterface {
        }
        return new Foobar() {
            public int foo() {
                return 42;
            }
        };
    }
}
```

Reply:
Also, for our use case, do lambdas behave the same as anonymous classes for this use case?